logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/MLSClientProvider.kt

```diff
@@ -1,20 +1,58 @@
 package com.wire.kalium.logic.data.client
 
+import com.wire.kalium.cryptography.CryptoQualifiedClientId
+import com.wire.kalium.cryptography.CryptoUserID
 import com.wire.kalium.cryptography.MLSClient
+import com.wire.kalium.cryptography.MLSClientImpl
+import com.wire.kalium.cryptography.MlsDBSecret
 import com.wire.kalium.logic.CoreFailure
 import com.wire.kalium.logic.data.conversation.ClientId
 import com.wire.kalium.logic.data.user.UserId
 import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.util.SecurityHelper
 import com.wire.kalium.persistence.kmm_settings.KaliumPreferences
+import com.wire.kalium.util.FileUtil
 
 interface MLSClientProvider {
-
     fun getMLSClient(clientId: ClientId? = null): Either<CoreFailure, MLSClient>
-
 }
 
-expect class MLSClientProviderImpl(
-    rootKeyStorePath: String,
-    userId: UserId,
-    clientRepository: ClientRepository,
-    kaliumPreferences: KaliumPreferences) : MLSClientProvider
+class MLSClientProviderImpl(
+    private val rootKeyStorePath: String,
+    private val userId: UserId,
+    private val clientRepository: ClientRepository,
+    private val kaliumPreferences: KaliumPreferences
+) : MLSClientProvider {
+
+
+    override fun getMLSClient(clientId: ClientId?): Either<CoreFailure, MLSClient> {
+        val currentClientId = clientId ?: clientRepository.currentClientId().fold({ return Either.Left(it) }, { it })
+
+        val location = "$rootKeyStorePath/${currentClientId.value}".also {
+            FileUtil.mkDirs(it)
+        }
```

Comment:
I think this would be a strong candidate to be absorbed by the incoming Okio refactoring. Maybe add a comment to merge into Kalium file system once that ticket is merged?